packages/react-components/react-persona/Spec.md

```diff
@@ -1,63 +1,312 @@
+
+- Note: v8 Persona only shows and icon if there's no presence and size is `tiny`, `size8`, or `size16`.
+
+_PersonaPresenceBadge vs PersonaCoin_: PersonaCoin allows you to have only a PresenceBadge when the size of the coin is `tiny`, `size8`, or `size16`. In v9's Persona you are able to render the PersonaPresenceBadge alone.
+
+_Image vs PersonaCoin_: PersonaCoin allows you to have an image only within the Avatar. This can be avoided by rendering a custom coin, but there's still all the dom and styling overhead that comes with PersonaCoin. v9's Persona allows you use a single image and does not resize the image.
+
+### Text lines
+
+Other than styling and naming, the text lines in Persona remain the same.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+![Avatar](./etc/images/avatar.png)
+
+Persona with Avatar:
+
+```jsx
+<Persona media={<PersonaAvatar name="Kevin Sturgis" />} primaryText="Kevin Sturgis" secondaryText="Software Engineer" />
+```
+
+![Avatar and PresenceBadge](./etc/images/avatar_badge.png)
+
+Persona with Avatar + PresenceBadge:
+
+```jsx
+<Persona
+  media={<PersonaAvatar name="Kevin Sturgis" badge={{ status: 'offline', outOfOffice: true }} />}
+  primaryText="Kevin Sturgis"
+  secondaryText="Software Engineer"
+  tertiaryText="Offline"
+/>
+```
+
+![Icon](./etc/images/icon.png)
+
+Persona with icon:
+
+```jsx
+<Persona
+  media={
+    <PersonaIcon>
+      <PersonCallRegular />
+    </PersonaIcon>
+  }
+  primaryText="Person Call Icon"
+/>
+```
+
+![PresenceBadge](./etc/images/badge.png)
+
+Persona with PresenceBadge:
+
+```jsx
+<Persona media={<PersonaPresenceBadge status="offline" outOfOffice={true} />} primaryText="Kevin Sturgis" />
+```
+
+![Image](./etc/images/image.png)
+
+Persona with image:
+
+```jsx
+<Persona
+  primaryText="Katri Athokas"
+  secondaryText="Software Engineer"
+  tertiaryText="CXE"
+  quaternaryText="Offline"
+  image={<img src="https://fabricweb.azureedge.net/fabric-website/assets/images/avatar/KatriAthokas.jpg" />}
+/>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+There are three alignment variants:
+
+- start: Media on the left and text on the right.
+- center: Media on top and text on the bottom.
+- end: Media on the right and text on the left.
+
+There are 4 Media variants:
+
+- PersonaPresenceBadge
+- PersonaIcon
+- PersonaAvatar
+- Image
+
+PersonaPresenceBadge, PersonaIcon, and PersonaAvatar are needed for the sizing behavior. PersonaContextProvider is provided by Persona and then consumed by these sub-components to update the persona's media state. This media state allows Persona to correctly style the text slots based on the media's type and size. See API section for more details on the context structure.
```

Comment:
Thanks for clarifying it 👍

I am wondering if `PersonaIcon` is really needed:

```tsx
function App() {
  return (
    <>
      {/* We can't apply sizing to sized icons */}
      <Persona media={< AccessTime24Filled />} />
      {/* For unsized we can specify `fontSize` to scale them */}
      <Persona media={< AccessTimeFilled />} />
    </>
   )
}
```

WDYT about it?

---

@ling1726 AFAIR you mentioned a similar problem with `Avatar` & `Table` sizing. Did you figure out a solution?

Reply:
The main thing PersonaIcon does is to add a font size and let Persona know that the media is an icon. If we were to remove PersonaIcon, we then drop support of the sizing styles for icons as well. Only Avatars and PresenceBadges would have the sizing behavior. 

I think adding a disclaimer about sized icons vs non-sized icons would be better in this case. I think it's worth having the sizing functionality for the icons if possible.

Reply:
> The main thing PersonaIcon does is to add a font size and let Persona know that the media is an icon. If we were to remove PersonaIcon, we then drop support of the sizing styles for icons as well

Can we specify `fontSize` in styles for `media` slot?

Reply:
if that's the case, I still need to know if the media type is an icon. While I could size it with a selector like `& > svg`, I still need to know that it's an icon to size the text lines correctly.

Reply:
> if that's the case, I still need to know if the media type is an icon. While I could size it with a selector like `& > svg`, I still need to know that it's an icon to size the text lines correctly.

Sorry, but I didn't get why... Is something except avatar, image, badge and icon can be specified in `media` slot?

Reply:
> What do you think?

TBH I am a bit lost at this point and it will be easier for me to understand the implementation. Will it be okay to keep going as it is for now and then follow up on this based on the implementation?

Reply:
Sounds good, I'll merge this PR #24704 and make a draft PR with the implementation! Sorry for the confusion, but I agree that it'll all make sense in the Implementation.

Reply:
@layershifter one last major change, I swear 😥😥 I've updated the spec to reflect a discussion we had offline. The new trjectory of this component will be to have a strict persona that aligns with v8's persona for partners. This will ensure that we have the same persona across products. 

For the extra functionality that included images, icons and other content, we are exploring two options but leaning towards option 1:
1. We will add a recipes section to the docs and give guidance on building a media object with the components we currently have.
2. We will provide building blocks for a MediaObject and make it flexible for the user.

Thanks so much for being attentive to this PR and I really appreciate all the feedback you've provided!

Reply:
Sorry, but it does not seem to be aligned with the spec in Figma. Complicating usage for images/icons does not seem for like a futureproof approach 🤔 

---

IMO the latest changes also return the problem with priorities and mutually exclusive slots:

```tsx
<Persona
  avatar={{ name: 'Kevin Sturgis', badge: { status: 'online' } }}
  presence={{ status: 'offline', outOfOffice: true }}
 />
```

For me, it's the same case as `appearance` with `Button`: we don't want users to do `<Button secondary primary />`, so we have `<Button appearance="primary" />`. This follows the principle "do the right thing by default".

https://github.com/microsoft/fluentui/blob/e0a0ab1481d0fb64f2a9804376ad17e4f4f8ea01/packages/react-components/react-button/src/components/Button/Button.types.ts#L17-L27